src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,118 @@
+package mate.academy.dao.impl;
+
+import mate.academy.dao.BookDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final int FIRST_PARAMETER_INDEX = 1;
+    private static final int SECOND_PARAMETER_INDEX = 2;
+    private static final int THIRD_PARAMETER_INDEX = 3;
+    private String request;
+
+    @Override
+    public Book create(Book book) {
+        request = "INSERT INTO books (title, price) VALUES(?, ?)";
+        Book savedBook = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(
+                     request, PreparedStatement.RETURN_GENERATED_KEYS
+             )) {
+            statement.setString(FIRST_PARAMETER_INDEX, book.getTitle());
+            statement.setObject(SECOND_PARAMETER_INDEX, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException(
+                        "Expected to insert at leas one row, but inserted 0 rows.");
+            }
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                Long id = resultSet.getObject(FIRST_PARAMETER_INDEX, Long.class);
+                savedBook = findById(id).orElseThrow();
```

Comment:
I`m sorry, if it is required to make another query to retrive actual data it is not needed. So please rollback this)))